src/app/qgsmeasuretool.cpp

```diff
@@ -201,26 +228,55 @@ void QgsMeasureTool::canvasReleaseEvent( QgsMapMouseEvent *e )
 {
   const QgsPointXY point = e->snapPoint();
 
-  if ( mDone ) // if we have stopped measuring any mouse click restart measuring
+  if ( mDone ) // If we have stopped measuring any mouse click restart measuring
   {
     mDialog->restart();
   }
 
-  if ( e->button() == Qt::RightButton ) // if we clicked the right button we stop measuring
+  if ( e->button() == Qt::RightButton ) // If we clicked the right button we stop measuring
   {
     mDone = true;
-    mRubberBand->removeLastPoint();
-    mDialog->removeLastPoint();
+    if ( mMeasureRadius )
+    {
+      // Add point to points rb
+      addPoint( point );
+    }
+    else
+    {
+      mRubberBand->removeLastPoint();
+      mDialog->removeLastPoint();
+    }
   }
   else if ( e->button() == Qt::LeftButton )
   {
     mDone = false;
+    if ( mMeasureRadius )
+    {
+      // Consecutive left clicks will clear the rubber bands and restart measuring
+      mPoints.clear();
+      mRubberBandPoints->removeLastPoint();
+      mRubberBand->removeLastPoint();
+      mRubberBandCircle->reset( Qgis::GeometryType::Polygon );
+    }
     addPoint( point );
   }
 
   mDialog->show();
 }
 
+QgsGeometry QgsMeasureTool::createCircleGeom()
+{
+  QgsPoint tmpCntrPt = QgsPoint( mPoints.at( 0 ) );
+  QgsPoint tmpOutrPt = mRubberBand->asGeometry().vertexAt( 1 );
+  double dist = mRubberBand->asGeometry().length();
+  double az = tmpCntrPt.azimuth( tmpOutrPt );
```

Comment:
Is azimuth really expected? By default, Z will have the same default value. Maybe you can snap, but it's never shown in the dialog. I doubt that you expect to have a particular oriented circle here.